Applications/Spire/Source/Ui/ListItem.cpp

```diff
@@ -8,23 +8,50 @@ using namespace boost::signals2;
 using namespace Spire;
 using namespace Spire::Styles;
 
+namespace {
+  auto DEFAULT_STYLE() {
+    auto style = StyleSheet();
+    style.get(Any()).
+      set(BackgroundColor(QColor::fromRgb(0xFF, 0xFF, 0xFF))).
+      set(border(scale_width(1), QColor::fromRgb(0, 0, 0, 0))).
+      set(horizontal_padding(scale_width(8)));
+    style.get(Hover()).set(
+      BackgroundColor(QColor::fromRgb(0xF2, 0xF2, 0xFF)));
+    style.get(Focus()).set(
+      border_color(QColor::fromRgb(0x4B, 0x23, 0xA0)));
+    style.get(Selected()).set(
+      BackgroundColor(QColor::fromRgb(0xE2, 0xE0, 0xFF)));
+    return style;
+  }
+}
+
 ListItem::ListItem(QWidget* component, QWidget* parent)
-    : QWidget(parent) {
+    : QWidget(parent),
+      m_is_selected(false) {
   setFocusPolicy(Qt::StrongFocus);
   auto layout = new QHBoxLayout(this);
   layout->setContentsMargins({});
-  m_button = new Button(component, this);
+  auto box = new Box(component);
+  box->setDisabled(true);
```

Comment:
Where in the spec is this specified?

Reply:
If the box is not disabled, the sub-component in the box will get focus even though the component is disabled. For example, when the component is a disabled and read-only TextBox, the TextBox still can get focus by pressing Tab key for some reason.

Reply:
Okay, that sounds good, look at fixing the issue with the ```Box```'s focus proxy.

Reply:
I wrote a simple demo using Qt Creator to simulate the case: The Button class with the focus policy Qt::StrongFocus, and the Box class with the focus proxy on a disabled QLineEdit. The disabled QLineEdit still can get focus if the Button contains the Box and the Box holds the QLineEdit, just like the ListItem. I believe that using the Qt::StrongFocus policy in the Button and the focus proxy in the Box together could cause the problem. When either not using the Button (no Qt::StrongFocus policy) or using the Button but removing setFocusProxy from the Box (no focus proxy), the disabled QLineEdit can't get focus.

Reply:
Okay this is a good catch, is there a proposal for how to modify ```Box``` or ```Button``` or some other modification to fix this issue?

Reply:
I don't prefer to modify Box or Button. I try to fix the issue by changing the focus proxy of the ListItem. When the sub-component is disabled, the ListItem's focus proxy is button and the box's focus proxy is none to prevent the disabled component from getting focus; when the sub-component is enabled, it will be set to the focus proxy of the ListItem, for example, there are two sub-components in the CheckBpx (a Icon and a TextBox), if the Checkbox is not the focus proxy, the focus will be on the button first and then on the Icon in turn by pressing Tab continually.